### PR TITLE
Test attributes deserve their own section

### PR DESCRIPTION
**Pull request by Havvy**
cc @QuietMisdreavus and @steveklabnik for possibly out-of-scope documentation here.

### CODE REVIEW

**Comment by Havvy** [comment]
Do not merge both this and the previous PR at the same time until I've edited the other PR to remove the "function-only attributes" section since the two PRs remove all of the attributes from that section.

**Comment by Havvy** [comment]
With the other PR merged, this one now also handles removing the entirety of the "Function only items" section and can be merged.

Hmm, I probably should have done that in a new commit, but it's 2am and I'm not thinking straight. But wanted to get this unblocked before I head to bed. Sorry. The only change since the previous comment is just the removal of that header.

File: src/attributes.md [hunk 1]
```diff
@@ -228,6 +220,42 @@ are transformed into `doc` attributes.
...
+> Note: The implementation of which return types are allowed is determined by
+> the unstable [`Termination`] trait.
+
+<!-- If the previous section needs updating (from "must take no arguments"
+  onwards, also update it in the crates-and-source-files.md file -->
+
+> Note: The test harness is ran by passing the `--test` argument to `rustc` or
+> using `cargo test`.
+
+Tests that return `()` pass as long as they terminate do not panic. Tests that
```
Comment (matthewjasper) [inline src/attributes.md]: missing 'and' after 'terminate'

Comment (Havvy) [reply src/attributes.md]: Fixed.

**Comment by matthewjasper** [comment]
Thanks!

**Pull request merged**